src/util/ConfigManager/ConfigManager.h

```diff
@@ -180,7 +180,7 @@ class ConfigManager {
   @return A reference to the newly created configuration option. This reference
   will stay valid, even after adding more options.
   */
-  template <SameAsAnyTypeInAvailableConfigOptionTypes OptionType>
+  template <ConfigOptionImpl::SupportedConfigOptionType OptionType>
```

Comment:
Either you can use the same namespace for both `ConfigOption` and `ConfigManager` , 
or you put a `using namespace ConfigOptionImpl` into this namespace here. Otherwise you don't gain that much here wrt shortness. But maybe a single `ConfigManagerImpl` namespace for everything suffices, as these classes are very entangled.

Reply:
Done, see [bd4eb95](https://github.com/ad-freiburg/qlever/pull/1214/commits/bd4eb9577b5b2f92aec344c88d0a6f8d44f8b5aa).